src/sgame/sg_bot_util.cpp

```diff
@@ -557,6 +557,28 @@ AINodeStatus_t BotActionEvolve( gentity_t *self, AIGenericNode_t* )
 	return STATUS_FAILURE;
 }
 
+// Ideally this function would be part of equipment_t<upgrade_t>::unlocked(),
+// but unfortunately we don't have access to the calling entity there
+static bool BotSkillArmorIsDesirable( const gentity_t *self, upgrade_t armor )
+{
+	if ( self->botMind->botSkillSet[BOT_H_BUY_MODERN_ARMOR] ) {
+		return true;
+	}
+
+	// let's pretend that we have one less armor rank unlocked:
+	switch ( armor )
+	{
+	case UP_BATTLESUIT:
+		return false;
+	case UP_LIGHTARMOUR:
```

Comment:
This could be split into two separate cases instead of doing some hacky math on the enum

Reply:
hmm, ok, the line count change isn't significant enough I guess